File: Random/selectionSort.c
#include <stdio.h>

int main() {
  int arr[] = {1, 1, 1, 1, 0, 5, 8};
  int len = sizeof(arr) / sizeof(arr[0]);

  for (int i = 0; i < len; i++) {
    int minIndex = i;
    for (int j = minIndex; j < len; j++) {
      if (arr[j] < arr[minIndex]) minIndex = j;
    }

    int temp = arr[minIndex];
    arr[minIndex] = arr[i];
    arr[i] = temp;
  }

  printf("[");
  for (int i = 0; i < len; i++) {
    printf("%d, ", arr[i]);
  }
  printf("]");
};

File: DSA with JS/Algorithm/Sorting/implementation/selection-sort.js
let arr = [65, 25, 12, 22, 11, 23, 28, 28, 77, 25, 77, 38, 20];
//         12, 65, 25, 22,  

for (let i = 0; i < arr.length; i++) {
  for (let j = i + 1; j < arr.length; j++) {
    if (arr[i] > arr[j]) {
      [arr[i], arr[j]] = [arr[j], arr[i]];
    }
  }
}
console.log(arr);
